Applications/SimpleShot/task_runner.cpp

```diff
@@ -213,10 +417,29 @@ int main(int argc, char **argv) {
 
   std::string model_str(argv[1]);
   std::string app_path =
-    argc == 6 ? std::string(argv[5]) : simpleshot::getcwd_();
+    argc == 7 ? std::string(argv[6]) : simpleshot::getcwd_();
+
   std::string method = argv[2];
   std::string train_path = app_path + "/tasks/" + argv[3];
   std::string val_path = app_path + "/tasks/" + argv[4];
+  std::string label_path;
+
+  if (std::strcmp(std::string(argv[5]).c_str(), "null") == 0) {
```

Comment:
Could you explain this?

Reply:
It is like when you don't add any new classes i.e. you don't have any label.txt file, while running the simpleshot runner I will provide string "null" in -place of label.txt file.

Reply:
After building, run with
$./executable model method train_file validation_file label_file app_path

3 . For Adding 5 Classes
/usr/lib/nntrainer/bin/applications/simpleshot_runner conv4 L2N facenet 5classtrain.dat 5classtest.dat 5classlabel.txt /usr/lib/nntrainer/bin/applications/res/app/SimpleShot

This appends centroid of 5 classes in tensor.bin and corresponding labels in AllLabels.txt

For Adding 2 new Classes

usr/lib/nntrainer/bin/applications/simpleshot_runner facenet L2N 2classtrain.dat 2classtest.dat 2classlabel.txt /usr/lib/nntrainer/bin/applications/res/app/SimpleShot

Reply:
Now see since you have added 7 classes now and there is not further addition of classes, So now suppose you have predict the class of already existing classes , So just provide the label path as "null" so it will directly call predict function instead of getCentroids

Reply:
Testing for Query Images belonging to 7 Classes
Give labels_file : 'null'
Train_file: could be anything doesn't ,matter
Test File : Query Images

/usr/lib/nntrainer/bin/applications/simpleshot_runner facenet L2N 7classtrain.dat 7classtest.dat null /usr/lib/nntrainer/bin/applications/res/app/SimpleShot

This gives out accuracy of 92.562%.